text/0143-versioning-and-stability.md

```diff
@@ -0,0 +1,164 @@
+* Semantic Conventions are a single package with a single version number.
+* Each contrib package has it's own version.
+
+Exception: in some languages, package managers may react poorly to experimental packages having a version higher than 0.X. In these cases, a language-specific workaround is required. Go, Ruby, and Javascript are examples.
+
+Note: different language implementations do not need to have matching version numbers, nor do implementations have to match the version of the specification they implement. For example, it is fine to have opentelemetry-python-api at 1.2.8, opentelemetry-java-api at 1.3.2, and the spec at 1.1.1.
+
+**Major version bump**
+Major version bumps only occur when there is a breaking change to a stable interface, or the removal of deprecated signals.
+
+OpenTelemetry values long term support. The expectation is that we will version to v1.0 once the first set of packages are declared stable. OpenTelemetry will then remain at v1.0 for years. There are no plans for a v2.0 of OpenTelemetry at this time. Additional stable packages, such as metrics and logs, will be added as minor version bumps.
+
+**Minor version bump**
+Most changes to OpenTelemetry result in a minor version bump.
+
+* New backward-compatible functionality added to any component.
+* Breaking changes to internal SDK components.
+* Breaking changes to experimental signals.
+* New experimental packages are added.
+* Experimental packages become stable.
+
+**Patch version bump**
+Patch versions make no changes which would require recompilation or potentially break application code. The following are examples of patch fixes.
+
+* Bug fixes which don't require minor version bump per rules above.
+* Security fixes.
+* Documentation.
+
+Currently, OpenTelemetry does NOT have plans to backport bug and security fixes to prior minor versions. Security and bug fixes are only applied to the latest minor version. We are committed to making it feasible for end users to stay up to date with the latest version of OpenTelemetry.
+
+## Long Term Support
+
+![long term support](img/0143_long_term.png)
+
+### API support
+
+Major versions of the API will be supported for a minimum of **three years** after the release of the next major API version. Support covers the following areas.
+
+API stability, as defined above, will be maintained.
+
+A version of the SDK which supports the last major version of the API will continue to be maintained during this period. Bug and security fixes will be backported. Additional feature development is not guaranteed.
+
+Contrib packages available when the API is versioned will continue to be maintained for the duration of this period. Bug and security fixes will be backported. Additional feature development is not guaranteed.
+
+### SDK Support
+
+SDK stability, as defined above, will be maintained for a minimum of **one year** after after the release of the next major SDK version.
```

Comment:
What are the practical implications of 3yrs for API and 1yr for the rest? In other words, how useful is API if its corresponding SDK is no longer supported (even for security fixes)?

Reply:
Here is how I intend it to work:

The three year API support explicitly requires that a working version of the SDK, contrib, and semantic conventions remain available and stable for the duration of the API LTS.

The SDK LTS is meant to cover SDK support when the API hasn't versioned. I expect the SDK to require breaking changes more frequently than the API.

So, as long as we don't break the API, our LTS guarantee is one year. If we break the API, LTS goes up to three years for a version of all components required to maintain the final version of the prior API.

Does that make sense?